flow/project.py

```diff
@@ -580,9 +580,54 @@ def __str__(self):
     def __call__(self, *jobs):
         """Return the command formatted with the supplied job(s)."""
         cmd = self._cmd(*jobs) if callable(self._cmd) else self._cmd
-        format_arguments = {"jobs": jobs}
-        if len(jobs) == 1:
-            format_arguments["job"] = jobs[0]
+        # Inconclusive test if cmd is a format string. If {} with anything in the interior is found
```

Comment:
One way we could conclusively test if the string is formatted is by applying the format operation and seeing if it modifies the string. e.g.
```python
cmd_input = "arbitrary string"
formatted = cmd_input.format(**format_arguments)
if cmd_input != formatted:
    warnings.warn("Formatting is deprecated.", FutureWarning)
```

Reply:
Good suggestion.